lib/routes/ups/track.ts

```diff
@@ -0,0 +1,107 @@
+import { Route } from '@/types';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+import puppeteer from '@/utils/puppeteer';
+
+export const route: Route = {
+    path: '/track/:trackingNumber',
+    categories: ['other'],
+    example: '/ups/track/1Z78R6790470567520',
+    parameters: { trackingNumber: 'The UPS tracking number (e.g., 1Z78R6790470567520).' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['ups.com/track?loc=en_US&tracknum=:trackingNumber'],
+            target: '/ups/track/:trackingNumber',
+        },
+    ],
+    name: 'UPS Tracking',
+    maintainers: ['Aquabet'],
+    handler,
+};
+
+async function handler(ctx) {
+    const { trackingNumber } = ctx.req.param();
+    const url = `https://www.ups.com/track?loc=en_US&tracknum=${trackingNumber}`;
+
+    const browser = await puppeteer();
+    const page = await browser.newPage();
+
+    await page.setRequestInterception(true);
+
+    // skip loading images, stylesheets, and fonts
+    page.on('request', (request) => {
+        if (['image', 'stylesheet', 'font'].includes(request.resourceType())) {
+            request.abort();
+        } else {
+            request.continue();
+        }
+    });
+
+    await page.goto(url, { waitUntil: 'domcontentloaded' });
+
+    await page.waitForSelector('tr[id^="stApp_ShpmtProg_LVP_progress_row_"]', {
+        timeout: 30000,
+    });
+
+    const content = await page.content();
+    browser.close();
+    const $ = load(content);
+
+    // Extract tracking events
+    const items = $('tr[id^="stApp_ShpmtProg_LVP_progress_row_"]')
```

Comment:
It would be better if you can intercept the response of `/track/api/Track/GetStatus` since it provides 22 shipment progress activities instead of 6 only (from `/ups/track/1Z78R6790470567520`)

Reply:
Using the official UPS API `/track/api/Track/GetStatus` requires OAuth 2.0, which involves account registration and a payment method, so I had to rely on scraping to fetch updates.

Reply:
Double-check your devtools network logs, this request does show up when you visit the site and I believe it doesn't require OAuth